src/ferny/session.py

```diff
@@ -153,6 +153,11 @@ async def connect(self,
             assert os.path.exists(self._controlsock)
             self._process = process
         except BaseException:
+            # If we get here because the InteractionResponder raised an
+            # exception then SSH might still be running, and may even attempt
+            # further interactions (ie: 2nd attempt for password).  We already
+            # have our exception and don't need any more info.  Kill it.
+            process.kill()
```

Comment:
Is that guaranteed to run still? I.e.could this fail with ESRCH?

Reply:
No.  It's internally double-guarded, first with the "did the process already exit?" check and then with a `except` around the actual `kill()` call, in case of a race.